### PR TITLE
feature: webpack config

### PR DESCRIPTION
**Pull request by Kupov**
https://jira.railsc.ru/browse/GOODS-401

### CODE REVIEW

**Comment by Kupov** [comment]
@dontuse

File: src/store.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {createStore, compose} from 'redux';
+import rootReducer from './reducers';
+
+export default function configureStore(initialState) {
+  const store = createStore(
+    rootReducer,
+    initialState,
+    compose(
+      window.devToolsExtension ? window.devToolsExtension() : f => f,
+  ));
```
Comment (apress-hound) [inline src/store.js]: Unexpected ')'.<br>Expected an identifier and instead saw ')'.<br>Expected ')' and instead saw ';'.<br>Missing semicolon.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+loadFont({
+  name: 'open-sans-light',
+  path: '/fonts/open-sans-light.woff.css'
+});
+
+ReactDOM.render(
+  <Provider store={store()}>
+    <App />
+  </Provider>,
```
Comment (apress-hound) [inline src/index.js]: Unclosed regular expression.<br>Unrecoverable syntax error. (90% scanned).

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+});
+
+loadFont({
+  name: 'open-sans-light',
+  path: '/fonts/open-sans-light.woff.css'
+});
+
+ReactDOM.render(
+  <Provider store={store()}>
+    <App />
```
Comment (apress-hound) [inline src/index.js]: Expected '}' to match '{' from line 25 and instead saw '<'.<br>Missing semicolon.<br>Expected an identifier and instead saw '>'.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  path: '/fonts/open-sans-bold.woff.css'
+});
+
+loadFont({
+  name: 'open-sans-light',
+  path: '/fonts/open-sans-light.woff.css'
+});
+
+ReactDOM.render(
+  <Provider store={store()}>
```
Comment (apress-hound) [inline src/index.js]: Expected an identifier and instead saw '<'.<br>Expected ')' and instead saw 'Provider'.<br>Missing semicolon.<br>'function closure expressions' is only available in Mozilla JavaScript extensions (use moz option).<br>Expected an identifier and instead saw '}'.<br>Expected an operator and instead saw '>'.

File: src/export.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import loadFont from './utils/font-loader';
+
+export loadFont;
```
Comment (apress-hound) [inline src/export.js]: Unexpected 'loadFont'.<br>Missing semicolon.

File: server.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+  app.use(express.static(__dirname + '/'));
+})();
+
+app.get(/.*/, function root(req, res) {
+  res.sendFile(__dirname + '/index.html');
+});
+
+const server = http.createServer(app);
+server.listen(process.env.PORT || 8080, function onListen() {
```
Comment (apress-hound) [inline server.js]: 'process' is not defined.

File: server.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+  app.use(require('webpack-hot-middleware')(compiler, {
+    log: console.log, path: '/__webpack_hmr', heartbeat: 10 * 1000,
+  }));
+
+  app.use(express.static(__dirname + '/'));
+})();
+
+app.get(/.*/, function root(req, res) {
+  res.sendFile(__dirname + '/index.html');
```
Comment (apress-hound) [inline server.js]: '__dirname' is not defined.

File: server.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+  app.use(require('webpack-dev-middleware')(compiler, {
+    noInfo: true, publicPath: webpackConfig.output.publicPath,
+  }));
+
+  app.use(require('webpack-hot-middleware')(compiler, {
+    log: console.log, path: '/__webpack_hmr', heartbeat: 10 * 1000,
+  }));
+
+  app.use(express.static(__dirname + '/'));
```
Comment (apress-hound) [inline server.js]: '__dirname' is not defined.

File: server.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+(function initWebpack() {
+  const webpack = require('webpack');
+  const webpackConfig = require('./webpack.config');
+  const compiler = webpack(webpackConfig);
+
+  app.use(require('webpack-dev-middleware')(compiler, {
+    noInfo: true, publicPath: webpackConfig.output.publicPath,
+  }));
+
+  app.use(require('webpack-hot-middleware')(compiler, {
```
Comment (apress-hound) [inline server.js]: 'require' is not defined.

File: server.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
+const http = require('http');
+const express = require('express');
+const app = express();
+
+(function initWebpack() {
+  const webpack = require('webpack');
+  const webpackConfig = require('./webpack.config');
+  const compiler = webpack(webpackConfig);
+
+  app.use(require('webpack-dev-middleware')(compiler, {
```
Comment (apress-hound) [inline server.js]: 'require' is not defined.

File: .babelrc [hunk 11]
```diff
@@ -0,0 +1,4 @@
+{
+  "plugins": ["transform-class-properties"],
```
Comment (dontuse) [inline .babelrc]: это нам зачем, уже есть же стейдж 2  https://github.com/tc39/proposal-class-public-fields

**Pull request merged**